0.10.6: build fails with new libxml 2.12.0

Here are compile time errors:
<details>

```console
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/snapper-0.10.6/snapper'
/bin/sh ../libtool  --tag=CXX   --mode=compile /usr/bin/g++ -DHAVE_CONFIG_H -I. -I..  -I/usr/include/libxml2   -DCONF_DIR='"/etc/sysconfig"' -D_FILE_OFFSET_BITS=64 -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c -o libsnapper_la-XmlFile.lo `test -f 'XmlFile.cc' || echo './'`XmlFile.cc
libtool: compile:  /usr/bin/g++ -DHAVE_CONFIG_H -I. -I.. -I/usr/include/libxml2 -DCONF_DIR=\"/etc/sysconfig\" -D_FILE_OFFSET_BITS=64 -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c XmlFile.cc  -fPIC -DPIC -o .libs/libsnapper_la-XmlFile.o
XmlFile.cc: In constructor 'snapper::XmlFile::XmlFile(int, const std::string&)':
XmlFile.cc:43:48: error: 'XML_PARSE_NOBLANKS' was not declared in this scope
   43 |         : doc(xmlReadFd(fd, url.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |                                                ^~~~~~~~~~~~~~~~~~
XmlFile.cc:43:69: error: 'XML_PARSE_NONET' was not declared in this scope
   43 |         : doc(xmlReadFd(fd, url.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |                                                                     ^~~~~~~~~~~~~~~
XmlFile.cc:43:15: error: 'xmlReadFd' was not declared in this scope; did you mean 'xmlRef'?
   43 |         : doc(xmlReadFd(fd, url.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |               ^~~~~~~~~
      |               xmlRef
XmlFile.cc: In constructor 'snapper::XmlFile::XmlFile(const std::string&)':
XmlFile.cc:53:51: error: 'XML_PARSE_NOBLANKS' was not declared in this scope
   53 |         : doc(xmlReadFile(filename.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |                                                   ^~~~~~~~~~~~~~~~~~
XmlFile.cc:53:72: error: 'XML_PARSE_NONET' was not declared in this scope
   53 |         : doc(xmlReadFile(filename.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |                                                                        ^~~~~~~~~~~~~~~
XmlFile.cc:53:15: error: 'xmlReadFile' was not declared in this scope
   53 |         : doc(xmlReadFile(filename.c_str(), NULL, XML_PARSE_NOBLANKS | XML_PARSE_NONET))
      |               ^~~~~~~~~~~
make[2]: *** [Makefile:686: libsnapper_la-XmlFile.lo] Error 1
/bin/sh ../libtool  --tag=CXX   --mode=compile /usr/bin/g++ -DHAVE_CONFIG_H -I. -I..  -I/usr/include/libxml2   -DCONF_DIR='"/etc/sysconfig"' -D_FILE_OFFSET_BITS=64 -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c -o libsnapper_la-Logger.lo `test -f 'Logger.cc' || echo './'`Logger.cc
libtool: compile:  /usr/bin/g++ -DHAVE_CONFIG_H -I. -I.. -I/usr/include/libxml2 -DCONF_DIR=\"/etc/sysconfig\" -D_FILE_OFFSET_BITS=64 -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -std=c++11 -Wall -Wextra -Wformat -Wnon-virtual-dtor -Wno-unused-parameter -Wsuggest-override -c Logger.cc  -fPIC -DPIC -o .libs/libsnapper_la-Logger.o
Logger.cc:151:5: error: 'xmlGenericErrorFunc' does not name a type
  151 |     xmlGenericErrorFunc xml_error_func_ptr = &xml_error_func;
      |     ^~~~~~~~~~~~~~~~~~~
Logger.cc: In function 'void snapper::initDefaultLogger()':
Logger.cc:172:38: error: 'xml_error_func_ptr' was not declared in this scope; did you mean 'xml_error_func'?
  172 |         initGenericErrorDefaultFunc(&xml_error_func_ptr);
      |                                      ^~~~~~~~~~~~~~~~~~
      |                                      xml_error_func
Logger.cc:172:9: error: 'initGenericErrorDefaultFunc' was not declared in this scope
  172 |         initGenericErrorDefaultFunc(&xml_error_func_ptr);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:735: libsnapper_la-Logger.lo] Error 1
```
</details>

--- a/snapper/Logger.cc
+++ b/snapper/Logger.cc
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <libxml/tree.h>
+#include <libxml/xmlerror.h>
 #include <string>
 #include <boost/thread.hpp>
 

--- a/snapper/XmlFile.cc
+++ b/snapper/XmlFile.cc
@@ -23,6 +23,7 @@
 
 #include <cstring>
 #include <unistd.h>
+#include <libxml/parser.h>
 
 #include "snapper/Exception.h"
 #include "snapper/XmlFile.h"

